src/thor/costmatrix.cc

```diff
@@ -256,10 +256,105 @@ bool CostMatrix::SourceToTarget(Api& request,
     uint32_t target_idx = connection_idx % target_location_list.size();
     uint32_t source_idx = connection_idx / target_location_list.size();
 
-    // first recost and form the path, if desired (either time and/or geometry requested)
-    const auto shape = RecostFormPath(graphreader, best_connection, source_location_list[source_idx],
-                                      target_location_list[target_idx], source_idx, target_idx,
-                                      time_infos[source_idx], invariant, shape_format);
+    std::string shape;
```

Comment:
small change, large diff: I pulled the path construction out of `RecostFormPath` (now that I spell it out, maybe this should be renamed to something like `RecostFormShape`), because we may need it for the headings and correlated lat/lon's. I figured this was better than the alternative, which would require passing a lot more information into the method, and making it do things beyond its original scope (i.e. forming part of the response).

Reply:
what if we kept this stuff down in FormPath but maybe we rename it to somethign else? Im actually ok with it still being named FormPath really, its the stuff you have to do after you do the expansion to make use of the paths that were found. in the case of other routing algorithms we just return the path itself (edgeids) but for matrix we are pretty much done so doing a bit of "serialization" is ok too. seems like we can just send the request API object to this method and modify it there to save on arguments to the method.

Reply:
I ended up doing what you suggested: leave the name as is and just change the signature of `RecostFormPath` to include the request so that we can set those new fields directly in there.